x-pack/filebeat/input/http_endpoint/input.go

```diff
@@ -0,0 +1,277 @@
+	if r.Header.Get("Content-Type") != "application/json" {
+		return http.StatusUnsupportedMediaType, in.createErrorMessage("Wrong Content-Type header, expecting application/json")
+	}
+
+	if r.Header.Get("Accept") != "application/json" {
+		return http.StatusNotAcceptable, in.createErrorMessage("Wrong Accept header, expecting application/json")
+	}
+	return 0, ""
+}
+
+// Validate if headers are current and authentication is successful
+func (in *HttpEndpoint) validateAuth(w http.ResponseWriter, r *http.Request) (uint, string) {
+	if in.config.Username == "" || in.config.Password == "" {
+		return http.StatusUnauthorized, in.createErrorMessage("Username and password required when basicauth is enabled")
+	}
+
+	username, password, _ := r.BasicAuth()
+	if in.config.Username != username || in.config.Password != password {
+		return http.StatusUnauthorized, in.createErrorMessage("Incorrect username or password")
+	}
+
+	return 0, ""
+}
+
+// Validates that body is not empty, not a list of objects and valid JSON
+func (in *HttpEndpoint) validateBody(w http.ResponseWriter, r *http.Request) (uint, string) {
+	if r.Body == http.NoBody {
+		return http.StatusNotAcceptable, in.createErrorMessage("Body cannot be empty")
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return http.StatusInternalServerError, in.createErrorMessage("Unable to read body")
+	}
+
+	isObject := in.isObjectOrList(body)
+	if isObject == "list" {
+		return http.StatusBadRequest, in.createErrorMessage("List of JSON objects is not supported")
+	}
+
+	objmap := make(map[string]interface{})
+	err = json.Unmarshal(body, &objmap)
+	if err != nil {
+		return http.StatusBadRequest, in.createErrorMessage("Malformed JSON body")
+	}
+
+	in.eventObject = &objmap
+
+	return 0, ""
+}
+
+// Ensure only valid HTTP Methods used
+func (in *HttpEndpoint) validateMethod(w http.ResponseWriter, r *http.Request) (uint, string) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed, in.createErrorMessage("Only POST requests supported")
+	}
+
+	return 0, ""
+}
+
+func (in *HttpEndpoint) createErrorMessage(r string) string {
+	return fmt.Sprintf(`{"message": "%v"}`, r)
+}
+
+func (in *HttpEndpoint) isObjectOrList(b []byte) string {
+	obj := bytes.TrimLeft(b, " \t\r\n")
+	if len(obj) > 0 && obj[0] == '{' {
+		return "object"
+	}
+
+	if len(obj) > 0 && obj[0] == '[' {
+		return "list"
+	}
+
+	return ""
+}
+
+func (in *HttpEndpoint) sendResponse(w http.ResponseWriter, h uint, b string) {
+	w.WriteHeader(int(h))
+	w.Write([]byte(b))
+}
+
+// Validates incoming requests and sends a new event upon success
+func (in *HttpEndpoint) apiResponse(w http.ResponseWriter, r *http.Request) {
+	// This triggers both validation and event creation
+	status, err := in.sendEvent(w, r)
+	if err != "" || status != 0 {
+		in.sendResponse(w, status, err)
+		return
+	}
+
+	in.sendResponse(w, http.StatusOK, in.config.ResponseBody)
```

Comment:
is this supposed to be `in.config.ResponseCode` rather than `http.StatusOK`?

Reply:
That is correct. This is something that I should have picked up. Initially it was put there since I found no good way to convert an int/uint to a StatusType, still looking into that.